pkg/spec/spec.go

```diff
@@ -174,10 +174,18 @@ func (config *CreateConfig) createConfigToOCISpec(runtime *libpod.Runtime, userM
 	}
 
 	hostname := config.Hostname
-	if hostname == "" && (config.NetMode.IsHost() || config.UtsMode.IsHost()) {
-		hostname, err = os.Hostname()
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to retrieve hostname")
+	if hostname == "" {
+		if utsCtrID := config.UtsMode.Container(); utsCtrID != "" {
+			utsCtr, err := runtime.GetContainer(utsCtrID)
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to retrieve hostname from dependency container %s", utsCtrID)
+			}
+			hostname = utsCtr.Hostname()
+		} else if config.NetMode.IsHost() || config.UtsMode.IsHost() {
+			hostname, err = os.Hostname()
+			if err != nil {
+				return nil, errors.Wrap(err, "unable to retrieve hostname")
+			}
 		}
 	}
```

Comment:
If  you get to line 199 and don't have a hostname still, should we be spitting out an error or perhaps a logrus.Debugf at least?

Reply:
We could. it's a fairly expected case, and I think runc overwrites the spec and adds mrscalloway as the hostname. At most I'd want to do a debug, but I'm not that inclined to. WDYT

Reply:
I'd be hip with a debug, that way if runc ever changes out from under us, we'd have a slightly easier way to debug.

Reply:
added, WDYT

Reply:
I think you worked a semi into an error message.  Bonus points!  +1
LGTM